__tests__/app.test.js

```diff
@@ -164,5 +159,160 @@ describe("post/api/articles/:articleid/comments", () => {
       });
   });
 });
+test("returns 404 error if article id not a number", () => {
+  const comment = {
+    username: "icellusedkars",
+    body: "Great article!",
+  };
+  return request(app)
+    .post("/api/articles/alex/comments")
+    .send(comment)
+    .expect(404)
+    .then(({ body }) => {
+      expect(body).toEqual({ msg: "404, article not found" });
+    });
+});
```

Comment:
This shouldnt be a 404 error. A 404 error should only be given when the id is valid but simply does not exist, in this case the id alex is simply invalid and should be a 400 instead.